File: Arrays and Strings/MeetingRooms.java
class Solution {
    public boolean canAttendMeetings(int[][] intervals) {
        Arrays.sort(intervals,(a,b)->{
            return a[0]-b[0];
        });

        for(int i = 1; i<intervals.length; i++) {
            int lo = intervals[i][0];
            int hi = intervals[i-1][1];
            //int prev = intervals[i][1];
            if(hi > lo) {
                return false;
            }
        }
        return true;
    }
}

// time complexity : O(nlogn)
// space complexity : O(1)

File: c++/meeting_rooms.cc
class Solution {
public:
    bool canAttendMeetings(vector<vector<int>>& intervals) {
       if (intervals.size() == 0 || intervals.size() == 1) {
           return true;
       }
       std::sort(intervals.begin(), intervals.end());
       for (int i = 1; i < intervals.size(); i++) {
           if (intervals[i][0] < intervals[i-1][1]) {
               return false;
           }
       }
       return true; 
    }
};
